src/confd/src/infix-if-lag.c

```diff
@@ -0,0 +1,162 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#include <fnmatch.h>
+#include <stdbool.h>
+#include <jansson.h>
+#include <arpa/inet.h>
+#include <net/if.h>
+
+#include <srx/common.h>
+#include <srx/lyx.h>
+#include <srx/srx_val.h>
+
+#include "ietf-interfaces.h"
+
+/*
+ * The lag-port noode is only in the dif at initial creation, or removal.
+ * When a lag is recreated, e.g., when changing mode, the lag-port is only
+ * in the cif.
+ *
+ * To simplify confd dependency handling we always call this function, so
+ * a lag member always has its lag master reset.  This is cruicial in the
+ * case described above when the the lag is removed to change mode.
+ */
+int lag_gen_ports(struct dagger *net, struct lyd_node *dif, struct lyd_node *cif, FILE *ip)
+{
+	const char *ifname = lydx_get_cattr(cif, "name");
+	struct lyd_node *node;
+	const char *lagname;
+	int err = 0;
+
+	node = lydx_get_descendant(lyd_child(dif), "lag-port", NULL);
+	if (node) {
+		struct lydx_diff lagdiff = { 0 };
+
+		if (!lydx_get_diff(lydx_get_child(node, "lag"), &lagdiff))
+			goto fail;
+
+		if (lagdiff.old) {
+			FILE *fp;
+
+			fp = dagger_fopen_current(net, "exit", lagdiff.old, 55, "exit.ip");
+			if (!fp) {
+				err = -EIO;
+				goto fail;
+			}
+
+			fprintf(fp, "link set %s nomaster\n", ifname);
+			fclose(fp);
+
+			return 0;
+		}
+
+		lagname = lagdiff.new;
+	} else {
+		node = lydx_get_descendant(lyd_child(cif), "lag-port", NULL);
+		lagname = lydx_get_cattr(node, "lag");
+		if (!node || !lagname)
+			goto fail; /* done, nothing to do */
+	}
+
+	fprintf(ip, "link set %s down master %s\n", ifname, lagname);
+
+	/* We depend on lag to exist before we can set master ... */
+	err = dagger_add_dep(net, ifname, lagname);
+	if (err)
+		ERROR("%s: unable to add dep to %s", ifname, lagname);
```

Comment:
Just like when detaching the port from the LAG, attaching needs to happen in the init action of the LAG, not in the LAG port. Have a look att `gen_join_leave()` in the new `infix-if-bridge-port.c` - I think you can almost copy-paste it to use for LAG ports as well.

Then: you can reverse the dependency. I.e., the LAG depends on all of its ports being setup before we can create the LAG itself.

Reply:
Ahaa, now I finally understand what you were talking about last week!  Yeah, that makes much more sense when the lag does the attaching. :+1: